Lib/Inc/CUartCom.h

```diff
@@ -11,25 +11,41 @@
 #ifndef CUARTCOM_H_
 #define CUARTCOM_H_
 
+#include <queue>
 #include <string>
+#include "IComChannel.h"
 #include "usart.h"
 
-#define UART_TIMEOUT 100
+#define UART_TIMEOUT   100
+#define MAX_RX_BUF_LEN 100
+#define MAX_QUEUE_SIZE 5
 
-class CUartCom
+class CUartCom : public IComChannel
 {
-private:
-    UART_HandleTypeDef *mp_huart;
-    GPIO_TypeDef *m_uart_de_port = nullptr;
-    uint16_t m_uart_de_pin;
-
 public:
-    CUartCom(UART_HandleTypeDef *p_huart);
-    CUartCom(UART_HandleTypeDef *p_huart,
+    CUartCom(const std::string name, UART_HandleTypeDef *p_huart);
+    CUartCom(std::string name,
+             UART_HandleTypeDef *p_huart,
              GPIO_TypeDef *uart_de_port,
```

Comment:
There is a CGpioWrapper class, so you should make use of it as much as possible. Code reuse is a useful thing to do.